sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -371,12 +373,14 @@ object DataType {
       byName: Boolean,
       resolver: Resolver,
       context: String,
+      storeAssignmentPolicy: StoreAssignmentPolicy.Value,
```

Comment:
maybe pass in a boolean flag `isStrict`?

Reply:
I think keeping the original policy is also fine. Otherwise, it is hard to tell that we are using ANSI mode if `isStrict` is `false`.